src/serialize/root.zig

```diff
@@ -0,0 +1,133 @@
+const std = @import("std");
+const testing = std.testing;
+const Allocator = std.testing.allocator;
+
+const MAX_FIELD_COUNT: u8 = 255;
+
+pub fn tomlize(allocator: std.mem.Allocator, obj: anytype, writer: anytype) !void {
+    try serializeStruct(allocator, obj, writer);
+}
+
+fn serializeStruct(allocator: std.mem.Allocator, value: anytype, writer: anytype) !void {
+    const ttype = @TypeOf(value);
+    const tinfo = @typeInfo(ttype);
+    if (!std.mem.eql(u8, @tagName(tinfo), "struct")) @panic("non struct type given to serialize");
+    const fields = comptime getFields(tinfo);
+
+    comptime var i: u8 = 0;
+    inline while (i < fields.len) {
+        const field = fields.buffer[i];
+        try serializeField(allocator, @typeInfo(field.type), field.name, @field(value, field.name), writer);
+        _ = try writer.write("\n");
+        i += 1;
+    }
+}
+
+fn serializeSimpleValue(allocator: std.mem.Allocator, t: std.builtin.Type, value: anytype, writer: anytype) !void {
```

Comment:
nit: in `zig-master` there is now `std.io.AnyWriter`. You can use it instead of `anytype`.

Reply:
I don't know about that because you still need to pass the type info about the underlying writing type in the definition of `AnyWriter`.

Reply:
Not sure I understood. You can get `AnyWriter` by calling `.any()` on the underlying writer. And by replacing `anytype` with `AnyWriter` gives more type control. I think it's new since `0.14.x`.